src/System.Management.Automation/engine/MshObject.cs

```diff
@@ -597,56 +638,65 @@ internal static PSObject ConstructPSObjectFromSerializationInfo(SerializationInf
-        /// adapted object. If an object is not adapted, this will be null.
+        /// Members from the adapter of the object before it was serialized
+        /// Null for live objects but not null for deserialized objects.
         /// </summary>
-        /// <remarks>
-        /// If an object is not adapted, InternalAdapter will use the dotnet adapter.
-        /// So there is no point falling back to the same dotnet adapter.
-        ///
-        /// If an object is adapted, this adapter will be used to resolve the dotnet
-        /// members.
-        /// </remarks>
-        internal Adapter InternalBaseDotNetAdapter => InternalAdapterSet.DotNetAdapter;
+        internal PSMemberInfoInternalCollection<PSPropertyInfo> adaptedMembers;
 
         /// <summary>
-        /// This is the adapter set that will contain the adapter of the baseObject
-        /// and the ultimate .net member lookup adapter.
-        /// See <see cref="PSObject.AdapterSet"/> for explanation.
+        /// Members from the adapter of the object before it was serialized
+        /// Null for live objects but not null for deserialized objects.
         /// </summary>
-        private AdapterSet InternalAdapterSet
-        {
-            get
-            {
-                if (_adapterSet == null)
-                {
-                    lock (_lockObject)
-                    {
-                        if (_adapterSet == null)
-                        {
-                            _adapterSet = GetMappedAdapter(_immediateBaseObject, GetTypeTable());
-                        }
-                    }
-                }
+        internal PSMemberInfoInternalCollection<PSPropertyInfo> clrMembers;
 
-                return _adapterSet;
-            }
-        }
+        private PSObjectFlags _flags;
+        private WriteStreamType _writeStream;
 
-        private AdapterSet _adapterSet;
+        #endregion instance fields
+
+        internal static PSTraceSource memberResolution = PSTraceSource.GetTracer("MemberResolution", "Traces the resolution from member name to the member. A member can be a property, method, etc.", false);
+
+        private static readonly ConditionalWeakTable<object, ConsolidatedString> s_typeNamesResurrectionTable = new ConditionalWeakTable<object, ConsolidatedString>();
 
-        internal bool hasGeneratedReservedMembers;
+        private static readonly Collection<CollectionEntry<PSMemberInfo>> s_memberCollection = GetMemberCollection(PSMemberViewTypes.All);
+        private static readonly Collection<CollectionEntry<PSMethodInfo>> s_methodCollection = GetMethodCollection();
+        private static readonly Collection<CollectionEntry<PSPropertyInfo>> s_propertyCollection = GetPropertyCollection(PSMemberViewTypes.All);
+
+        internal static readonly DotNetAdapter dotNetInstanceAdapter = new DotNetAdapter();
+        private static readonly DotNetAdapter s_baseAdapterForAdaptedObjects = new BaseDotNetAdapterForAdaptedObjects();
```

Comment:
I'd recommend revising this section so that all `private` and `internal` members are grouped separately, and organised by type and probably alphabetically. There're quite a few members here, so organising this will make it a lot easier to work with in future.

Reply:
I think static/instance is more important to group by than accessibility. That is what will show us the object layout. I can make the fields private, and make an internal getter for the fields that need to be accessed internally.

Reply:
We'll disable CodeFactor rules for such sorting in  #8500 - fill free to feedback.